aviatrix/resource_aviatrix_edge_equinix.go

```diff
@@ -371,16 +371,19 @@ func marshalEdgeEquinixInput(d *schema.ResourceData) *goaviatrix.EdgeEquinix {
 		interface2 := &goaviatrix.EdgeEquinixInterface{
 			IfName:       interface1["name"].(string),
 			Type:         interface1["type"].(string),
-			Bandwidth:    interface1["bandwidth"].(int),
 			PublicIp:     interface1["wan_public_ip"].(string),
 			Tag:          interface1["tag"].(string),
 			Dhcp:         interface1["enable_dhcp"].(bool),
 			IpAddr:       interface1["ip_address"].(string),
 			GatewayIp:    interface1["gateway_ip"].(string),
 			DnsPrimary:   interface1["dns_server_ip"].(string),
 			DnsSecondary: interface1["secondary_dns_server_ip"].(string),
-			VrrpState:    interface1["enable_vrrp"].(bool),
-			VirtualIp:    interface1["vrrp_virtual_ip"].(string),
+		}
+
+		// vrrp_state and virtual_ip are only applicable for LAN interfaces
+		if interface1["type"].(string) == "LAN" && interface1["enable_vrrp"].(bool) {
+			interface2.VrrpState = interface1["enable_vrrp"].(bool)
+			interface2.VirtualIp = interface1["vrrp_virtual_ip"].(string)
```

Comment:
Would it make sense to have a validation error if the interface type is not `"LAN"` and `vrrp_virtual_ip` is set?
Or if we want to remove any validation in Terraform, then rather forward it to the controller and have that reject it?
To me, it would be surprising if a property is silently dropped.

Reply:
Currently we have the check in the controller to throw an error if vrrp_state or virtual_ip is set for interfaces other than LAN so I didn't include it here.